File: data/derived_cot/rq1_traces/translation/qwen/java_to_py/code_output/RegexUtils.py
import re

class RegexUtils:

    def match(pattern, text):
        compiled_pattern = re.compile(pattern)
        return compiled_pattern.fullmatch(text) is not None

    def findall(pattern, text):
        compiled_pattern = re.compile(pattern)
        return list(compiled_pattern.findall(text))

    def split(pattern, text):
        compiled_pattern = re.compile(pattern)
        return compiled_pattern.split(text, maxsplit=-1)

    def sub(pattern, replacement, text):
        compiled_pattern = re.compile(pattern)
        return compiled_pattern.sub(replacement, text)

    def generateEmailPattern():
        return r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,}\b'

    def generatePhoneNumberPattern():
        return r'\b\d{3}-\d{3}-\d{4}\b'

    def generateSplitSentencesPattern():
        return r'[.!?][\s]{1,2}(?=[A-Z])'

    def splitSentences(text):
        pattern = RegexUtils.generateSplitSentencesPattern()
        return RegexUtils.split(pattern, text)

    def validatePhoneNumber(phoneNumber):
        pattern = RegexUtils.generatePhoneNumberPattern()
        return RegexUtils.match(pattern, phoneNumber)

    def extractEmail(text):
        pattern = RegexUtils.generateEmailPattern()
        return RegexUtils.findall(pattern, text)

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_cpp/RegexUtils.cpp
#include <regex>
#include <vector>
#include <string>
#include <iterator>

class RegexUtils {
public:
    bool match(const std::string& pattern, const std::string& text) {
        std::regex compiledPattern(pattern);
        return std::regex_match(text, compiledPattern);
    }

    std::vector<std::string> findall(const std::string& pattern, const std::string& text) {
        std::regex compiledPattern(pattern);
        std::sregex_iterator matcher(text.begin(), text.end(), compiledPattern);
        std::sregex_iterator end;
        std::vector<std::string> matches;
        while (matcher != end) {
            matches.push_back((*matcher).str());
            ++matcher;
        }
        return matches;
    }

    std::vector<std::string> split(const std::string& pattern, const std::string& text) {
        std::regex compiledPattern(pattern);
        std::sregex_token_iterator it(text.begin(), text.end(), compiledPattern, -1);
        std::sregex_token_iterator end;
        return std::vector<std::string>(it, end);
    }

    std::string sub(const std::string& pattern, const std::string& replacement, const std::string& text) {
        std::regex compiledPattern(pattern);
        return std::regex_replace(text, compiledPattern, replacement);
    }

    std::string generate_email_pattern() {
        return "\\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Z|a-z]{2,}\\b";
    }

    std::string generate_phone_number_pattern() {
        return "\\b\\d{3}-\\d{3}-\\d{4}\\b";
    }

    std::string generate_split_sentences_pattern() {
        return "[.!?][\\s]{1,2}(?=[A-Z])";
    }

    std::vector<std::string> splitSentences(const std::string& text) {
        std::string pattern = generate_split_sentences_pattern();
        return split(pattern, text);
    }

    bool validate_phone_number(const std::string& phoneNumber) {
        std::string pattern = generate_phone_number_pattern();
        return match(pattern, phoneNumber);
    }

    std::vector<std::string> extractEmail(const std::string& text) {
        std::string pattern = generate_email_pattern();
        return findall(pattern, text);
    }
};